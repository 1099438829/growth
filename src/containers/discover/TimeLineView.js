import React, { Component, PropTypes } from 'react';
import { View, Text } from 'react-native';
import Icon from '../../icon/GrowthFont';

import AppStyle from '../../theme/styles';

class TimeLine extends Component {
  static componentName = 'TimeLine';

  static propTypes = {
    selfStyle: View.propTypes.style,
    lineParentStyle: View.propTypes.style,
    lineIconStyle: Text.propTypes.style,
    topLineStyle: Text.propTypes.style,
    bottomLineStyle: Text.propTypes.style,
    cententParentStyle: Text.propTypes.style,
    title: PropTypes.string.isRequired,
    titleStyle: Text.propTypes.style,
    content: PropTypes.string.isRequired,
    contentStyle: Text.propTypes.style,
    hideTopLine: PropTypes.bool,
    itemIndex: PropTypes.number,
  };

  static defaultProps = {
    selfStyle: AppStyle.timeLine,
    lineParentStyle: AppStyle.timeLineParent,
    lineIconStyle: AppStyle.timeLineIcon,
    topLineStyle: AppStyle.timeLineTopLine,
    bottomLineStyle: AppStyle.timeLineBottomLine,
    cententParentStyle: AppStyle.timeLineCententParent,
    title: '',
    titleStyle: AppStyle.timeLineTitle,
    content: '',
    contentStyle: AppStyle.timeLineContent,
    hideTopLine: false,
    itemIndex: 0,
  };

  render = () => (
    <View style={this.props.selfStyle}>
      <View style={this.props.lineParentStyle}>
        <Text style={[this.props.topLineStyle, (this.props.hideTopLine || this.props.itemIndex === 0) ? { backgroundColor: 'transparent' } : {}]} />
        <Icon name={'circle'} size={20} style={this.props.lineIconStyle} />
        <Text style={this.props.bottomLineStyle} />
      </View>
      <View style={this.props.cententParentStyle}>
        <Text style={this.props.titleStyle}>{this.props.title}</Text>
        <Text style={this.props.contentStyle}>{this.props.content}</Text>
      </View>
    </View>
  );
}
export default TimeLine;
