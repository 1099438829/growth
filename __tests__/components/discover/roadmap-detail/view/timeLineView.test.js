import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import TimeLineView from '../../../../../src/components/discover/roadmap-detail/view/TimeLineView';


it('renders correctly', () => {
  const timeLine = renderer.create(
    <TimeLineView
      title={'title'}
      content={'23333333'}
      itemIndex={0}
      key={'key'}
    />,
  );
  const timeLineJson = timeLine.toJSON();
  expect(timeLineJson).toMatchSnapshot();
});
