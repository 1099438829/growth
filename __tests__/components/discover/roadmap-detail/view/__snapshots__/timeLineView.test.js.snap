// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "#03a9f4",
            "height": 20,
            "width": 2,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "borderRadius": 10,
            "color": "#03a9f4",
            "height": 20,
            "width": 20,
          },
          Object {
            "fontFamily": "growth",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î±­
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "#03a9f4",
          "flex": 1,
          "width": 2,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 5,
        "flex": 1,
        "marginBottom": 10,
        "marginRight": 15,
        "marginTop": 15,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#03a9f4",
          "flex": 1,
          "height": 30,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "paddingLeft": 15,
          }
        }
      >
        title
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "padding": 15,
        }
      }
    >
      23333333
    </Text>
  </View>
</View>
`;
